src/cast_core/src/CollectionScanner.cpp

```diff
@@ -227,9 +241,117 @@ void countScan(CollectionScanner::PhaseConfig* config,
     statTracker.success();
 }
 
+void pointInTimeScan(mongocxx::pool::entry& client,
+                     CollectionScanner::PhaseConfig* config,
+                     std::optional<bsoncxx::types::b_timestamp> readClusterTime) {
+    int64_t docCount = 0;
+    int64_t scanSize = 0;
+    bool scanFinished = false;
+    mongocxx::options::client_session sessionOption;
+    sessionOption.causal_consistency(false);
```

Comment:
you could probably construct the sessionOptions during PhaseConfig construction-time and re-use it but I'm not sure how this object's lifetime works

Reply:
This is only used in the pointInTimeScan and it is only called once per point-in-time scan phase. I think it is fine to keep it as is so the definition is closer to the code that uses it.